AlfrescoApp/Views/MultiSelect/MultiSelectContainerView.m

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+ * Copyright (C) 2005-2018 Alfresco Software Limited.
+ *
+ * This file is part of the Alfresco Mobile iOS App.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+
+#import "MultiSelectContainerView.h"
+
+static CGFloat const kMultiSelectAnimationDuration = 0.2f;
+
+@interface MultiSelectContainerView()
+
+@property (nonatomic, assign) BOOL subviewAdded;
+
+@end
+
+@implementation MultiSelectContainerView
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    self = [super initWithFrame:frame];
+    if(!self)
```

Comment:
the more readable version of overriding the init method would be:
`self = [super initWithFrame:frame]
  if (self) {[self createToolabar];}
  return self;`
that way if the super call fails when returning self you will be returning nil

Edit #2: I would create the toolbar component under the init method then, under layoutSubviews I would remove the subviewAdded check and replace it with the following:
`if (!self.toolbar.superview) {
 [self addSubview: self.toolbar];
}` i.e. you're removing the ivar and basing your checks on your view hierarchy to check if your toolbar is already added.